### PR TITLE
Fixes #14156 - Upgrade to Rails 4.2

### PR DESCRIPTION
**Pull request by daviddavis**
Requires the rails42 branch of foreman and https://github.com/theforeman/foreman-docker/pull/157

### CODE REVIEW

**Comment by domcleal** [comment]
The only incompatible change I see in this PR is the use of #deliver_now, which is only required to silence a deprecation.

I'd strongly suggest merging compatible changes into master and only making incompatible changes on a branch mirroring Foreman's rails42, or after 4.2 is merged to develop.  Life's simpler all around with compatible changes.

**Comment by daviddavis** [comment]
@domcleal sounds good. Thanks!

File: test/katello_test_helper.rb [hunk 1]
```diff
@@ -169,9 +169,9 @@ def set_user(user = nil)
 
   def get_organization(org = nil)
     saved_user = User.current
-    User.current = User.find(users(:admin))
+    User.current = User.find(users(:admin).id)
     org = org.nil? ? :empty_organization : org
-    organization = Organization.find(taxonomies(org.to_sym))
+    organization = Organization.find(taxonomies(org.to_sym).id)
```
Comment (ehelms) [inline test/katello_test_helper.rb]: These changes only work on Rails 4.2?

**Comment by daviddavis** [comment]
Broke out compatible Rails 4.2 changes into https://github.com/Katello/katello/pull/5958

**Comment by jlsherrill** [comment]
APJ

**Comment by daviddavis** [comment]
[test]

**Pull request merged**